addons/web/static/src/legacy/js/core/concurrency.js

```diff
@@ -202,7 +202,7 @@ return {
                         }
                         return returnedResult;
                     };
-                    Promise.resolve(result).then(always).guardedCatch(always);
```

Comment:
test (but maybe on non legacy mutex)

Reply:
done :-)
It passes on the new implem, and fails on the legacy one, without the fix